### PR TITLE
[SCI] add build file tree support via Cargo.toml dependency mappings

### PR DESCRIPTION
**Pull request by anderruiz**
## Summary

- Implement `GetArtifact` on `CargoLockExtractor` to parse `[package].name` from sibling `Cargo.toml` and discover internal project dependencies via `path = "..."` entries across `[dependencies]`, `[dev-dependencies]`, and `[build-dependencies]`
- Register `SimpleProcessor` for `FileTypeCargoToml` to wire Rust into the build file tree pipeline
- Add 8 test cases covering name extraction, path deps, missing files, deduplication, scan root boundaries, and edge cases

Follows the same pattern established in #179 (Go build file tree support) and #178 (.NET ProjectReference tracing).

## Test plan

- [x] `go test ./pkg/extractor/rust/...` passes (8 new tests)
- [x] `go test ./pkg/sbomgen/...` passes
- [x] `go test ./...` passes (full suite, no regressions)
- [ ] Verify CI passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/extractor/rust/cargo-toml-artifact.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+	var parsed CargoToml
+	if err := toml.Unmarshal(content, &parsed); err != nil {
+		return &models.ScannedArtifact{ArtifactDetail: models.ArtifactDetail{Filename: f.Path()}}, err
+	}
+
+	artifact := &models.ScannedArtifact{
+		ArtifactDetail: models.ArtifactDetail{
+			Name:      parsed.Package.Name,
+			Filename:  f.Path(),
```
[the remark this answers is not in the document]
Comment (anderruiz) [reply pkg/extractor/rust/cargo-toml-artifact.go]: Fixed. Changed `Filename` from `f.Path()` (Cargo.lock) to `cargoTomlFile.Path()` (Cargo.toml) so the artifact node in the dependency graph matches the `ProjectDeps` targets and `GetBuildFileTrees` can traverse the tree correctly.
